### PR TITLE
changelog: update the 2.5 `CHANGELOG.md` with missing entries

### PR DESCRIPTION
**Pull request by Ebaneck**
**Component**:

'changelog'

**Context**: 
- Update the `CHANGELOG.md` file with missing entries

**Summary**:

**Acceptance criteria**: 

---

Closes: #ISSUE_NUMBER

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,5 +2,76 @@
...
 
+### Breaking changes
+- Basic authentication has been deprecated in favour of OpenID Connect (OIDC)
+with Dex being the Identity provider of choice.
+This comes with the following:
+    - The Kubernetes API server authentication flag `--basic-auth` will be
+      removed and henceforth no longer be used in favour of `--oidc-issuer-url`.
+    - Fresh installations of MetalK8s will come with 2 Dex pods in a new
+      namespace `metalk8s-auth`
+    - Complete re-brand of the MetalK8s UI
```
Comment (gdemonet) [inline CHANGELOG.md]: Is this related to Dex in particular? Maybe worth its own item

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,5 +2,76 @@
...
+    - Authentication and access to MetalK8s UI and Grafana services will
+      henceforth implement OIDC based authentication
```
Comment (gdemonet) [inline CHANGELOG.md]: Let's focus on user facing changes in here, especially on what is breaking:

```suggestion
- Basic authentication has been deprecated in favour of OpenID Connect (OIDC)
  with Dex being deployed as a local Identity Provider, used by Kubernetes
  API and Grafana.
  
  This implies:
    - The existing users defined for Kubernetes API Basic Auth (in
      `/etc/htpasswd`) and for the Grafana admin will become unusable
    - A default admin user will be created in Dex, with the new
      credentials `admin@metalk8s.invalid`:`password`
    - Procedure to edit and add users can now be found
      [here](link to the docs page, maybe on RTD)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,5 +2,76 @@
...
+
+- A new framework for persisting Cluster and Services Configurations (CSC) has
+been added to ensure configurations set by administrators are not lost during
+upgrade or downgrade.
```
Comment (gdemonet) [inline CHANGELOG.md]: Link to the architecture document would be great :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -2,5 +2,76 @@
...
+This comes with the following:
+    - New CSC's to hold configurations for:
+        - Dex configurations `metalk8s-dex-config`
+        - Grafana configurations `metalk8s-grafana-config`
+        - Prometheus configurations `metalk8s-prometheus-config`
+        - Alertmanager configurations `metalk8s-alertmanager-config`
+    - Procedures for changing and applying new configuration values for each
+      CSC has been documented.
```
Comment (gdemonet) [inline CHANGELOG.md]: Let's rewrite it to make it more actionable (just a suggestion, feel free to reword):

```suggestion
User-provided configuration is now stored in ConfigMaps,
and MetalK8s tooling will honor the values provided when
deploying its services:
    - Dex uses `metalk8s-auth/metalk8s-dex-config`
    - Grafana uses `metalk8s-monitoring/metalk8s-grafana-config`
    - Prometheus uses `metalk8s-monitoring/metalk8s-prometheus-config`
    - Alertmanager uses `metalk8s-monitoring/metalk8s-alertmanager-config`
    
Documentation for changing and applying configuration values is
documented [here](...).
Note that any configuration applied on other Kubernetes objects
(e.g. a configuration Secret that Alertmanager uses, or the
Deployment of Grafana) will be lost on upgrade, and admins
should make sure to prepare the relevant ConfigMaps from their
existing configuration before upgrading to this version.
```

**Review by gdemonet — CHANGES_REQUESTED** [review]
Let's reword the breaking changes section a bit to make it easier for readers to know where to go after (esp. give links to the relevant pages in documentation, since we have them :wink:)

Otherwise, LGTM

File: CHANGELOG.md (hunk 2 above)
Comment (Ebaneck) [reply CHANGELOG.md]: Done

**Comment by slaperche-scality** [comment]
I dunno if you track bug fixes here, but [the one about storage-operator](https://github.com/scality/metalk8s/issues/2409) that just got merged may need to be mentioned (was done since 2.4 though).

**Comment by Ebaneck** [comment]
> I dunno if you track bug fixes here, but [the one about storage-operator](https://github.com/scality/metalk8s/issues/2409) that just got merged may need to be mentioned (was done since 2.4 though).

cc @alexandre-allard-scality

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,5 +2,91 @@
...
+      - Prometheus uses `metalk8s-monitoring/metalk8s-prometheus-config`
+      - Alertmanager uses `metalk8s-monitoring/metalk8s-alertmanager-config`
+
+  - Documentation for changing and applying configuration values is
+    found [here](https://metal-k8s.readthedocs.io/en/development-2.5/operation/cluster_and_service_configuration.html).
+
+    Note that any configuration applied on other Kubernetes objects
+    (e.g. a configuration Secret that Alertmanager uses, or the
+    Deployment of Grafana) will be lost upon upgrade, and admins
+    should make sure to **save** the relevant ConfigMaps from their
```
Comment (gdemonet) [inline CHANGELOG.md]: Isn't it a creation rather than a save?

Comment (Ebaneck) [reply CHANGELOG.md]: done

**Review by gdemonet — APPROVED** [review]
